### PR TITLE
Mod: add comment for metadata

### PR DESCRIPTION
**Pull request by watermelo**
**What this PR does**:
add comment for metadata

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: metadata/identifier/base_metadata_identifier.go [hunk 1]
```diff
@@ -47,7 +46,7 @@ func joinParams(joinChar string, params []string) string {
 	return joinedStr
 }
 
-// getIdentifierKey...
+// getIdentifierKey gets metadata identifier key by @params
```
Comment (watermelo) [inline metadata/identifier/base_metadata_identifier.go:49]: I didn't find its usage, Need big-old help.

**Comment by watermelo** [comment]
#570

File: metadata/identifier/service_metadata_identifier.go [hunk 2]
```diff
@@ -27,12 +27,12 @@ type ServiceMetadataIdentifier struct {
 	BaseMetadataIdentifier
 }
 
-// getIdentifierKey...
+// getIdentifierKey gets metadata identifier key by @mdi.protocol and @mdi.revision
 func (mdi *ServiceMetadataIdentifier) getIdentifierKey(params ...string) string {
-	return mdi.BaseMetadataIdentifier.getIdentifierKey(mdi.protocol + constant.KEY_REVISON_PREFIX + mdi.revision)
+	return mdi.BaseMetadataIdentifier.getIdentifierKey(mdi.protocol, constant.KEY_REVISON_PREFIX, mdi.revision)
```
Comment (zouyx) [inline metadata/identifier/service_metadata_identifier.go:32]: I think should not change `+` to `,` 

When
`
protocol="a"
revision="b"
`

In `BaseServiceMetadataIdentifier.getIdentifierKey` use `+` , result is `arevisionb`.

In `BaseServiceMetadataIdentifier.getIdentifierKey` use `,` , result is `a:revision:b`.

File: metadata/identifier/base_metadata_identifier.go [hunk 3]
```diff
@@ -25,19 +25,21 @@ import (
 	"github.com/apache/dubbo-go/common/constant"
 )
 
+// BaseMetadataIdentifier defined for describe the Metadata base identify
```
Comment (AlexStocks) [inline metadata/identifier/base_metadata_identifier.go:28]: describe --> description

File: metadata/identifier/base_metadata_identifier.go [hunk 4]
```diff
@@ -25,19 +25,21 @@ import (
...
+// BaseMetadataIdentifier is the base implement of BaseMetadataIdentifier interface
 type BaseServiceMetadataIdentifier struct {
 	serviceInterface string
 	version          string
 	group            string
 	side             string
 }
 
-// joinParams...
+// joinParams will join the specified char in slice, and build as string
```
Comment (AlexStocks) [inline metadata/identifier/base_metadata_identifier.go:42]: 'build as string' --> return a string

File: metadata/identifier/subscribe_metadata_identifier.go [hunk 5]
```diff
@@ -17,17 +17,18 @@
...
 package identifier
 
+// SubscriberMetadataIdentifier is inherit baseMetaIdentifier with service params: Revision
 type SubscriberMetadataIdentifier struct {
 	revision string
 	BaseMetadataIdentifier
 }
 
-// getIdentifierKey...
+// GetIdentifierKey will return string format as service:Version:Group:Side:Revision
```
Comment (AlexStocks) [inline metadata/identifier/subscribe_metadata_identifier.go:26]: getIdentifierKey's return string format is "service:Version:Group:Side:Revision"

**Pull request merged**